lib/authorizer/index.js

```diff
@@ -0,0 +1,202 @@
+var _ = require('lodash'),
+    sdk = require('postman-collection'),
+    Requester = require('../requester').Requester,
+
+    Authorizer;
+
+/**
+ * This class manages all authorizations.
+ *
+ * @param options
+ * @param {Boolean} options.interactive - Controls interactive vs non-interactive behavior.
+ * @param callback
+ * @constructor
+ */
+Authorizer = function (options, callback) {
+    var self = this;
+
+    self.interactive = options.interactive;
+
+    callback(null, this);
+};
+
+_.assign(Authorizer.prototype, {
+    /**
+     * Checks whether required parameters are present, and takes the appropriate next step.
+     *
+     * @param context
+     * @param run
+     * @param callback
+     */
+    preVerify: function (context, run, callback) {
+        // if stateless mode is enabled, returns affirmative
+        if (!this.interactive) {
+            return callback(null, true);
+        }
+
+        var handler = context.auth;
+
+        if (!handler) {
+            // fine, no authentication is to be performed.
+            return callback(null, true);
+        }
+
+        Requester.create({
```

Comment:
Are these `Requester`s clones of `run.requester`?

The reasoning for creating new Requesters in each of the `Authorizer.handler` is to not mutate the `run.requester` or the need to create new `Requester` with different options?